packages/cicero-core/src/template.js

```diff
@@ -172,15 +178,60 @@ class Template {
         return hasher.digest('hex');
     }
 
+    /**
+     * signs a string made up of template hash and time stamp using private key derived
+     * from the keystore
+     * @param {String} [p12File] - encoded string of p12 keystore file
```

Comment:
The `[foo]` syntax indicates that the parameter is optional. Is that correct here?

Reply:
I don't think so. Even I'm new to JSDoc, what I understood till now is that `[foo]` is just the name of the parameter for that particular method. If there is an optional parameter to be included then it'll be simply `[option]`. [Reference](https://github.com/accordproject/cicero/blob/01f5393dd3075512d688844773f7a3b871062135/packages/cicero-core/src/template.js#L182).

Reply:
I believe Matt is correct about [JSDoc](https://stackoverflow.com/questions/5873927/how-to-indicate-param-is-optional-using-inline-jsdoc)

`[p12File]` means `p12File` is an optional parameter of the function. If the parameter is mandatory you should simply write `@param {String} p12File`.